src/Features/Core/Portable/Diagnostics/DiagnosticEventTaskScheduler.cs

```diff
@@ -27,6 +27,7 @@ public DiagnosticEventTaskScheduler(int blockingUpperBound)
 
         private void Start()
         {
+            Thread.CurrentThread.Name = "Roslyn Diagnostics";
```

Comment:
can we just make it same pattern as the other one?

Reply:
The other instances I saw start a thread directly instead of using the Task factory. We could transition to the same method here and name the created thread, but I wasn't sure that was the right approach.

Reply:
let's go the other route since there we use both Thread and Task. I rather just use 1 (Thread or Task) rather mix them.